src/ndt-client.js

```diff
@@ -98,6 +115,64 @@ NDTjs.prototype.checkEnvironmentSupport = function() {
 };
 
 /**
+ * A generic login creation system for NDT.
+ *
+ * @param {Number} desiredTests The types of tests requested from the server
+ *  signaled based on a bitwise operation of the test ids.
+ * @returns {Uint8Array} An array of bytes suitable for sending on a binary
+ *  websocket.
+ */
+
+NDTjs.prototype.makeNDTLogin = function(desiredTests) {
+  var messageBody = '{ "msg": "v3.5.5", "tests": "' +
+      String(desiredTests | 16) + '" }';
+
+  return this.makeNDTMessageArray(this.constants.messageType.MSG_EXTENDED_LOGIN,
+      messageBody);
+};
+
+/**
+ * A generic message creation system for NDT.
+ *
+ * @param {Number} messageType The type of message according to NDT's
+ *  specification.
+ * @param {String} messageContent The message to send the server.
+ * @returns {Uint8Array} An array of bytes suitable for sending on a binary
+ *  websocket.
+ */
+
+NDTjs.prototype.makeNDTMessage = function(messageType, messageContent) {
+   var messageBody = '{ "msg": "' + messageContent + '" }';
```

Comment:
I had considered it and purposefully did not, firstly because of the compatibility question, but more so because I did not want to start to build objects and invoke JSON's stringify if I could simply concatenate a few strings. This might be premature optimization, but it's something that I want to tag as a feature and then experiment with (inc. JSON clean up in general) when the JS is able to conduct tests.